runhouse/resources/hardware/utils.py

```diff
@@ -711,3 +752,92 @@ def __str__(self):
         if self.retry:
             s += ", retry in {0}ms".format(self.retry)
         return s
+
+
+###################################
+# KUBERNETES SETUP
+###################################
+def setup_kubernetes(
+    namespace: Optional[str] = None,
+    kube_config_path: Optional[str] = None,
+    context: Optional[str] = None,
+    **kwargs,
+):
+    if kwargs.get("provider") and not kwargs.get("provider") == "kubernetes":
+        raise ValueError(
+            f"Recieved non kubernetes provider {kwargs.get('provider')} with kubernetes specific "
+            "cluster arguments."
+        )
+
+    if (
+        kwargs.get("server_connection_type")
+        and kwargs.get("server_connection_type") != ServerConnectionType.SSH
+    ):
+        raise ValueError(
+            "Runhouse K8s Cluster server connection type must be set to `ssh`. "
+            f"You passed {kwargs.getserver_connection_type}."
+        )
+
+    if context and namespace:
+        logger.warning(
+            "You passed both a context and a namespace. Ensure your namespace matches the one in your context.",
+        )
+
+    launcher = kwargs.get("launcher") or configs.launcher
+    if namespace and launcher == "local":
+        # Set the context only if launching locally
+        # check if user passed a user-defined namespace
+        cmd = f"kubectl config set-context --current --namespace={namespace}"
```

Comment:
just in terms of ordering, I think we should set context before namespace (where relevant), bc if the local kube config has multiple contexts we'll need to first have that set before setting the namespace